substrate/primitives/blockchain/src/tx_priority.rs

```diff
@@ -0,0 +1,108 @@
+use codec::{Decode, Encode};
+use serde::Deserialize;
+use sp_runtime::{
+    traits::Block as BlockT,
+    transaction_validity::TransactionPriority,
+};
+use sp_core::H160;
+
+/// Provides transaction details required by `TransactionPriorityModule`.
+pub trait TransactionDetailProvider: Send + Sync {
+    type Block: BlockT;
+    fn get_transaction_detail(&self, tx: <Self::Block as BlockT>::Extrinsic) -> Option<TransactionDetail>;
+}
+
+/// Implemented for the `Client`, to get tx priorities from the transaction pool.
+pub trait TransactionPriorityModuleT {
+    type Block: BlockT;
+    fn get_priority(&self, tx: <Self::Block as BlockT>::Extrinsic) -> Option<TransactionPriority>;
+}
+
+/// Main struct for getting TX priority from the specified list.
+pub struct TransactionPriorityModule<Block> {
+    priority_list: Vec<TransactionPriorityItem>,
+    // dynamic dispatch here. This can be changed to use a generic type, but this would require
+    // way more changes in the existing code.
+    pub tx_detail_provider: Box<dyn TransactionDetailProvider<Block = Block>>,
```

Comment:
as far as i understood the usage of this 

you do 

priority_module.tx_detai_provider.get_tx_detail

and then 

priority_module.get_priority

i wonder if the first can be hidden from the usage

so you would only do priorty_module.get_priority(tx) and within that, you would call the tx_detail_provider.